sourced/ml/transformers/content2ids.py

```diff
@@ -96,43 +95,58 @@ def build_mapping():
         return linguist2pygments
 
 
-class Content2Ids(Transformer):
+class IdentifiersToDataset(Transformer):
 
     def __init__(self, idfreq: bool, **kwargs):
         super().__init__(**kwargs)
         self.idfreq = idfreq
 
     def __call__(self, rows: RDD):
-        list_RDDs = []
         if self.idfreq:
-            for i in (0, 1):
-                # initial structure of x: (identifier, (repositoryId, filepath))
-                freq_processed = rows.map(lambda x: (x[0], x[1][i])).distinct()
-                list_RDDs.append(self.reduce_rows(freq_processed))
-            list_RDDs.append(self.reduce_rows(rows))
-            return rows \
-                .context.union(list_RDDs) \
-                .groupByKey() \
-                .mapValues(list) \
-                .map(lambda x: Row(
-                        token=x[0],
-                        token_split=" ".join(TokenParser(min_split_length=1).split(x[0])),
-                        num_repos=x[1][0],
-                        num_files=x[1][1],
-                        num_occ=x[1][2]))
+            return self.process_stats(rows)
         else:
-            return rows \
-                .map(lambda x: x[0]) \
-                .distinct() \
-                .map(lambda x: Row(
-                        token=x,
-                        token_split=" ".join(TokenParser(min_split_length=1).split(x))))
+            return self.process(rows)
 
     def reduce_rows(self, rows: RDD):
         return rows \
             .map(lambda x: (x[0], 1)) \
             .reduceByKey(operator.add)
 
+    def process_stats(self, rows: RDD):
+        """
+        Process rows to gather identifier frequencies.
+        num_repos is the number of repositories where the identifier appears in.
+        num_files is the number of files where the identifier appears in.
+        num_occ is the total number of occurences of the identifier.
+        """
+        list_RDDs = []
+        for i in (0, 1):
+            # initial structure of x: (identifier, (repositoryId, filepath))
+            freq_processed = rows.map(lambda x: (x[0], x[1][i])).distinct()
+            list_RDDs.append(self.reduce_rows(freq_processed))
+        list_RDDs.append(self.reduce_rows(rows))
+        return rows \
+            .context.union(list_RDDs) \
+            .groupByKey() \
+            .mapValues(list) \
+            .map(lambda x: Row(
+                    token=x[0],
+                    token_split=" ".join(TokenParser(min_split_length=1).split(x[0])),
+                    num_repos=x[1][0],
+                    num_files=x[1][1],
+                    num_occ=x[1][2]))
+
+    def process(self, rows: RDD):
+        """
+        Process rows to gather identifier without frequencies.
+        """
+        return rows \
+            .map(lambda x: x[0]) \
```

Comment:
You might use `.keys()` instead of `map(...)` here.

Reply:
True, thanks for your review!